File: frontend/src/api/system.ts
import request from './request'
import type { SystemStatus } from '../types/system'

export interface SystemInfo {
  version: string
}

export function getSystemInfo() {
  return request.get<never, SystemInfo>('/system/info')
}

export function getSystemStatus() {
  return request.get<never, SystemStatus>('/system/status')
}

File: backend/app/api/system.py
from fastapi import APIRouter

from app.core.version import get_app_version
from app.schemas.system import SystemInfoOut, SystemStatusOut
from app.services import system_status_service

router = APIRouter()


@router.get("/system/info", response_model=SystemInfoOut)
def system_info() -> SystemInfoOut:
    return SystemInfoOut(version=get_app_version())


@router.get("/system/status", response_model=SystemStatusOut)
def system_status() -> dict:
    return system_status_service.get_status()
